src/main/java/org/jpeek/Main.java

```diff
@@ -91,11 +102,50 @@ public static void main(final String... args) throws IOException {
      * @throws IOException If fails
      */
     private void run() throws IOException {
+        if (this.target.exists()) {
```

Comment:
@llorllale Well, this works, but it is very procedural. I think this logic should be inside an object called ``Target`` and it should be executed a few lines bellow, when calling ``this.target.toPath()``.

On the other hand, it's not your fault, the architecture headed towards this code from the beginning (``target`` was always a simple File, so all the logic about it is scattered around), so please, leave a puzzle for implementing such an object and refactoring this code :)